### PR TITLE
[LocalNet] Run Relayer and AppGateServer

### PR DESCRIPTION
**Pull request by okdas**
### Human Summary

As we've got relayer moved from the alpha repo, and appgateserver merged into main, we need to be able to run them on LocalNet. Support for this on helm-charts side have been merged.

### Summary generated by Reviewpad on 11 Nov 23 17:40 UTC
This pull request introduces changes related to the relayer and appgateserver.

It adds new values files for the appgateserver and relayer in the localnet/kubernetes directory and sets their enable values to true.

Additionally, it modifies the Tiltfile to include the appgateserver helm resource, which uses the new values file and sets the replicaCount based on the localnet configuration.

Overall, this patch adds support for the appgateserver and enables the relayer in the localnet setup.

## Type of change

Select one or more:

- [x] New feature, functionality or library
- [ ] Bug fix
- [ ] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

- [x] **Run all unit tests**: `make go_develop_and_test`
- [x] **Verify Localnet manually**: See the instructions [here](TODO: add link to instructions)

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, updated documentation and left TODOs throughout the codebase

### CODE REVIEW

**Review by Olshansk — COMMENTED** [review]
This LGTM. @okdas What do you see as the blocking next step to get this in?

**Comment by bryanchriswhite** [comment]
Based on the state of [charts/poktroll/templates/scripts.yaml on main](https://github.com/pokt-network/helm-charts/blob/main/charts/poktroll/templates/scripts.yaml), it looks like we still need to:
1. Update the subcommand name `relayer` -> `relayminer`
2. Decide when/how to workaround integrate with the solution to the supplier1/app1 vs. supplier2/app2 for the relayminer's and appgate server's signing-key flags, respectively. See [#pokrtoll in discord](https://discord.com/channels/824324475256438814/1138895490331705354/1173919224171860009) and https://github.com/pokt-network/poktroll/pull/177/files#r1392304589.
3. Ensure that the suppler and app both have public keys on chain.

File: Tiltfile [hunk 1]
```diff
@@ -127,11 +128,23 @@ helm_resource(
     poktroll_chart,
     flags=[
         "--values=./localnet/kubernetes/values-common.yaml",
+        "--values=./localnet/kubernetes/values-relayer.yaml",
```
Comment (h5law) [inline Tiltfile:131]: Screenshot 2023-11-14 at 19 44 45

Still getting this error with the relayers in tilt. Is it still looking for the old command and not the new one?

**Review by h5law — DISMISSED** [review]
AppGateServer works perfectly in localnet k8s! 🔥 Still not working for me with the relayerminer but once thats done this is gunna be so good!

File: localnet/kubernetes/values-relayer.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+relayer:
```
Comment (Olshansk) [inline localnet/kubernetes/values-relayer.yaml:1]: Per @bryanchriswhite's comment, we need to rename this to `relayminer`

**Comment by Olshansk** [comment]
@bryanchriswhite Agreed regarding `s/relayer/relayminer`  in this PR.

Re (2) & (3) - I believe it's outside the scope of this PR and we'll capture it in #180.

File: Tiltfile (hunk 1 above)
Comment (okdas) [reply Tiltfile:131]: I am surprised you still see that, as we no longer have the `pocketd` binary.

**Comment by okdas** [comment]
Per discussion above, I'm going to tweak the helm charts to use `relayminer` instead of `relayer`.

File: Tiltfile (hunk 1 above)
Comment (okdas) [reply Tiltfile]: Confirmed: this would not happen on a PR with `relayminer` implemented. We just need to merge everything.

**Comment by okdas** [comment]
I ended up splitting helm charts for `appgate-server` and `relayminer` - https://github.com/pokt-network/helm-charts/pull/6/files. The services are going to have different configs, ports exposed so it made sense to separate them. Also, this should potentially simplify key management for node runners. We will get to that point in a few weeks. :)

Just need a review on that PR and we should be good.

File: Dockerfile.dev [hunk 3]
```diff
@@ -19,5 +19,7 @@ RUN mv /poktroll/bin/poktrolld /usr/bin/poktrolld
 
 EXPOSE 8545
```
Comment (Olshansk) [inline Dockerfile.dev:20]: NIT: Adding a comment on what each exposed port is used for will go a long way!

**Review by Olshansk — APPROVED** [review]
Left one comment but LGTM otherwise

File: Dockerfile.dev (hunk 3 above)
Comment (okdas) [reply Dockerfile.dev:20]: @Olshansk these will be changed soon, I'll prepare a doc for us to discuss the configs/ports and dependencies for different actors.

File: Dockerfile.dev [hunk 4]
```diff
@@ -19,5 +19,7 @@ RUN mv /poktroll/bin/ignite /usr/bin/ && mv /poktroll/bin/poktrolld /usr/bin/
...
```
Comment (Olshansk) [inline Dockerfile.dev:20]:
```suggestion
# TODO_TECHDEBT(@okdas): Ports are not documented as they will soon be changed with a document to follow
EXPOSE 8545
```

**Review by Olshansk — APPROVED** [review]
@okdas Anything blocking from getting this in?

**Comment by okdas** [comment]
> @okdas Anything blocking from getting this in?

nope, let's get it in!

**Pull request merged**